src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -1220,6 +1220,136 @@ struct ZHighToZLowGRUOpLowering : public ConversionPattern {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// Lower ZHigh FixGRUY to Krnl
+//===----------------------------------------------------------------------===//
+
+struct ZHighToZLowFixGRUYOpLowering : public ConversionPattern {
+  ZHighToZLowFixGRUYOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
```

Comment:
I have a new implementation with better performance for this function. 
1. Reuse the input buffer, instead of creating a new buffer.
2. Change the loop structure. Iterate the batch first and load sequence_lens[bs]. Then the lower bound of the sequence loop will be min(sequence_lens[bs], 0). We will just fill the elements unconditionally with initial value.

The two optimizations are unrelated. I am testing them for performance. The current implement is correct and simple in code.

Reply:
Got it. Feel free to include it here or another PR.